multiple definition of `sockaddr`

**OS**: Manjaro 20.1
**Kernel**: 5.4.52-1-MANJARO
**GCC**: 10.1.0

i got this error on `make run-noaccel` while trying to build from 0e13e4f
```
ld: nanos/output/platform/pc/src/unix/socket.o:nanos/src/unix/socket.h:4: multiple definition of `sockaddr'; nanos/output/platform/pc/src/net/netsyscall.o:nanos/src/unix/socket.h:4
```

i made local changes on https://github.com/nanovms/nanos/blob/af2b48726915caaa953de3f26e0273bbf7d0fd0c/rules.mk#L66
to `CFLAGS+=    -Wall -Werror -Wno-char-subscripts -Wno-address-of-packed-member -Wno-array-bounds` to get around the following error
```
/nanos/platform/pc/service.c:595:49: error: array subscript 2147483648 is outside array bounds of ‘u8[1]’ {aka ‘unsigned char[1]’} [-Werror=array-bounds]
```
not sure if this is related.

--- a/platform/pc/service.c
+++ b/platform/pc/service.c
@@ -592,7 +592,7 @@ void init_service(u64 rdi, u64 rsi)
         u8 e820_entries = *(params + BOOT_PARAM_OFFSET_E820_ENTRIES);
         region e820_r = (region)(params + BOOT_PARAM_OFFSET_E820_TABLE);
         extern u8 END;
-        u64 kernel_size = u64_from_pointer(&END - KERNEL_BASE);
+        u64 kernel_size = u64_from_pointer(&END) - KERNEL_BASE;
         u64 *pdpt = 0;
         u64 *pdt = 0;
         for (u8 entry = 0; entry < e820_entries; entry++) {

--- a/src/unix/socket.h
+++ b/src/unix/socket.h
@@ -1,4 +1,4 @@
-struct sockaddr {
+typedef struct sockaddr {
     u16 family;
     u8 sa_data[14];
 } *sockaddr;

